tests/acceptance/request_test.py

```diff
@@ -35,16 +37,48 @@ def validation_context(request, response=None):
         exceptions.RequestValidationError,
         exceptions.ResponseValidationError,
         exceptions.PathNotFoundError,
-    ) as e:
+    ):
         raise exception_response(206)
-        assert e.child is not None
     except Exception:
         raise exception_response(400)
 
 
 validation_ctx_path = 'tests.acceptance.request_test.validation_context'
 
 
+def test_echo_date_with_pyramid_swagger_renderer(test_app):
+    input_object = {'date': datetime.date.today().isoformat()}
+
+    response = test_app.post_json('/echo_date', input_object)
+
+    # If the request is served via Swagger1.2
+    assert response.status_code == 200
+    assert response.json == input_object
+
+
+def test_echo_date_with_json_renderer(test_app):
+    today = datetime.date.today()
+    input_object = {'date': today.isoformat()}
+
+    exc = None
+    response = None
+    try:
+        response = test_app.post_json('/echo_date_json_renderer', input_object)
+    except TypeError as exception:
+        exc = exception
+
+    served_swagger_versions = test_app.app.registry.settings['pyramid_swagger.swagger_versions']
+
+    if '2.0' in served_swagger_versions:
```

Comment:
Do we need one more test for 2.0 where we use the new renderer? The one above is testing 1.2 (at least according to the comment)

Reply:
``test_app`` is generated by https://github.com/striglia/pyramid_swagger/blob/8cb3d30959006a85777374847b9aae8308dd98c3/tests/acceptance/request_test.py#L13-L15 and generates 3 tests with support for Swagger 1.2 only, Swagger 2.0 only and Swagger 1.2 and 2.0 

The idea is that in case Swagger 2.0 support is present we're going to receive an exception from ``JSON`` renderer because it is not able to properly render a ``datetime.date`` instance.
The case of ``Swagger1.2`` support only is different because in case of Swagger1.2 support only ``request.swagger_data`` is equivalent to ``request.json`` according to https://github.com/striglia/pyramid_swagger/blob/25fc16c9070bb39cec602b0374977e4bbd2c29a3/pyramid_swagger/tween.py#L337-L352
You can see that I'm asserting that the field is a string (https://github.com/macisamuele/pyramid_swagger/blob/maci-pyramid_swagger-renderer/tests/acceptance/app/__init__.py#L45)

Reply:
Ah, the parameterizing makes sense then